litellm/utils.py

```diff
@@ -516,6 +516,16 @@ def get_dynamic_callbacks(
         returned_callbacks.extend(dynamic_callbacks)  # type: ignore
     return returned_callbacks
 
+@lru_cache(None)
```

Comment:
we should have an upperbound on this right. e.g max size=128 

Don't want a memory leak